src/components/legend/style.js

```diff
@@ -30,6 +32,8 @@ module.exports = function style(s, gd, legend) {
     var fullLayout = gd._fullLayout;
     if(!legend) legend = fullLayout.legend;
     var constantItemSizing = legend.itemsizing === 'constant';
+    var itemWidth = legend.itemwidth;
+    var centerPos = (itemWidth + constants.itemGap * 2) / 2;
```

Comment:
Non-blocking: 
It would be nice if we calculate `centerTransform` 
```js
var centerTransform = 'translate(' + centerPos + ',0)';
```
and use it in multiple places down here e.g.
```js
.attr('transform', centerTransform);
```